src/java/org/apache/cassandra/locator/ReplicaPlans.java

```diff
@@ -333,32 +331,37 @@ static void assureSufficientLiveReplicas(Keyspace keyspace, ConsistencyLevel con
     @VisibleForTesting
     public static ReplicaPlan.ForTokenWrite forWrite(Keyspace keyspace, ConsistencyLevel consistencyLevel, EndpointsForToken natural, EndpointsForToken pending, Predicate<Replica> isAlive, Selector selector) throws UnavailableException
     {
-        return forWrite(keyspace, consistencyLevel, ReplicaLayout.forTokenWrite(natural, pending), isAlive, selector);
+        return forWrite(keyspace, consistencyLevel, ReplicaLayout.forTokenWrite(keyspace.getReplicationStrategy(), natural, pending), isAlive, selector);
```

Comment:
It seems that `ReadRepairTest.createRepairHandler` is calling this with a null keyspace, producing the NPE test failures that can be seen in the provided CI results. This can be fixed by using `readPlan.keyspace()` instead of `ks` in `AbstractReadRepairTest#repairPlan(ReplicaPlan.ForRangeRead, EndpointsForRange)`.